lib/src/source/hosted.dart

```diff
@@ -220,7 +221,11 @@ class BoundHostedSource extends CachedSource {
     try {
       // TODO(sigurdm): Implement cancellation of requests. This probably
       // requires resolution of: https://github.com/dart-lang/sdk/issues/22265.
-      bodyText = await httpClient.read(url, headers: pubApiHeaders);
+      bodyText = await withAuthenticatedClient(
+        systemCache,
```

Comment:
We need to refactor this a bit.. what if the `CredentialsStore` is a property on `SystemCache`?

Here we're loading a file and parsing JSON for every single outgoing request :see_no_evil: